core/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java

```diff
@@ -218,9 +220,6 @@ public void testFailPrimaryStartedCheckReplicaElected() {
         assertThat(routingTable.index("test").shard(0).primaryShard().currentNodeId(), anyOf(equalTo("node1"), equalTo("node2")));
         assertThat(routingTable.index("test").shard(0).replicaShards().size(), equalTo(1));
         assertThat(routingTable.index("test").shard(0).replicaShards().get(0).state(), equalTo(UNASSIGNED));
-
-        logger.info("fail the shard again, check that nothing happens");
```

Comment:
This is tested in ShardFailedClusterStateTaskExecutorTests. `testDuplicateFailuresAreOkay` checks duplicate failures (`createExistingShards` creates the duplicates using `int numberOfTasks = randomIntBetween(failures.size(), 2 * failures.size());`). `testNonExistentShardsAreMarkedAsSuccessful` fails a shard of non-existent index (`createNonExistentShards` uses existing shards with wrong allocation id, or wrong index etc.
In conclusion, the existing tests there should cover us.